pytensor/tensor/rewriting/basic.py

```diff
@@ -975,6 +976,30 @@ def local_join_make_vector(fgraph, node):
         return [ret]
 
 
+@register_specialize
+@register_canonicalize
+@register_useless
+@node_rewriter([Sum])
+def local_sum_make_vector(fgraph, node):
+    """A sum of a MakeVector node is just the sum of the elements."""
+    (array,) = node.inputs
```

Comment:
Does it make sense to do this if the vector is used elsewhere (multiple clients)? In that case it still has to be created anyways

Reply:
Good question...
I guess if we don't do a rewrite like this because of multiple uses, we could also just miss things where both uses can be rewritten.
I'm not sure which option is better.

Reply:
Yeah I was thinking that. Lets do it. Eggs would be good for this kind of stuff!